lib/mix/lib/mix/tasks/compile.ex

```diff
@@ -98,6 +98,42 @@ defmodule Mix.Tasks.Compile do
     |> maybe_prepend(:yecc)
   end
 
+  @doc """
+  Reenables given compilers so they can be executed again down the stack.
+
+  If an umbrella project reenables compilers, they are re-enabled for all
+  child projects.
+
+  By default reenables all compilers.
+  """
+  @spec reenable([{:compilers, compilers} | {:protocols, protocols}]) :: :ok
+        when compilers: :all | [atom()], protocols: :consolidate | nil
+  def reenable(opts \\ []) do
+    compilers =
+      case Keyword.get(opts, :compilers, :all) do
+        :all -> compilers()
+        list when is_list(list) -> list
+      end
+
+    compilers =
+      if Keyword.get(opts, :protocols) in [:consolidate, true] do
+        compilers
+        |> Enum.map(&to_string/1)
+        |> Enum.reduce([], fn
+          "elixir", acc -> ["protocols", "elixir" | acc]
+          "app", acc -> ["app", "protocols", "elixir" | acc]
```

Comment:
If `:elixir` has been given, append `:protocols` right after it, prepend both `:elixir` and `:protocols` to `:app` otherwise, so that `:app` is still the latest compiler.

I am not sure if this is actually needed.